### PR TITLE
Avoid calling warning() before `Errors` module initialised

### LINKED ISSUE #5396: [BUG] cythonize crashes when an include directory contains files with extra . characters
### Describe the bug

When cythonize is passed an `include_path` that contains a directory with the following files, it crashes in the following way

simon@tek-top /var/tmp/build-root/openSUSE_Tumbleweed-x86_64/home/abuild/rpmbuild/BUILD/python-efl-1.26.0/include ➤   ls
efl.c_eo.pxd               efl.ecore_input.pxd      efl.edje.pxd              efl.eo.pxd             efl.utils.conversions.pxd
efl.c_ethumb.pxd           efl.ecore.pxd            efl.eina.pxd              efl.ethumb_client.pxd  efl.utils.deprecated.pxd
efl.ecore_con.enums.pxd    efl.ecore_x.pxd          efl.elementary.enums.pxd  efl.ethumb.enums.pxd   efl.utils.logger.pxd
efl.ecore_con.pxd          efl.edje_edit.enums.pxd  efl.elementary.pxd        efl.ethumb.pxd
efl.ecore.enums.pxd        efl.edje_edit.pxd        efl.emotion.enums.pxd     efl.evas.enums.pxd
efl.ecore_input.enums.pxd  efl.edje.enums.pxd       efl.emotion.pxd           efl.evas.pxd

[    5s]   File "/home/abuild/rpmbuild/BUILD/python-efl-1.26.0/setup.py", line 304, in <module>
[    5s]     ext_modules = cythonize(
[    5s]   File "/usr/lib64/python3.10/site-packages/Cython/Build/Dependencies.py", line 985, in cythonize
[    5s]     module_list, module_metadata = create_extension_list(
[    5s]   File "/usr/lib64/python3.10/site-packages/Cython/Build/Dependencies.py", line 834, in create_extension_list
[    5s]     kwds = deps.distutils_info(file, aliases, base).values
[    5s]   File "/usr/lib64/python3.10/site-packages/Cython/Build/Dependencies.py", line 687, in distutils_info
[    5s]     return (self.transitive_merge(filename, self.distutils_info0, DistutilsInfo.merge)
[    5s]   File "/usr/lib64/python3.10/site-packages/Cython/Build/Dependencies.py", line 696, in transitive_merge
[    5s]     return self.transitive_merge_helper(
[    5s]   File "/usr/lib64/python3.10/site-packages/Cython/Build/Dependencies.py", line 708, in transitive_merge_helper
[    5s]     for next in outgoing(node):
[    5s]   File "Cython/Utils.py", line 129, in Cython.Utils.cached_method.wrapper
[    5s]   File "/usr/lib64/python3.10/site-packages/Cython/Build/Dependencies.py", line 609, in cimported_files
[    5s]     pxd_file = self.find_pxd(module, filename)
[    5s]   File "Cython/Utils.py", line 129, in Cython.Utils.cached_method.wrapper
[    5s]   File "/usr/lib64/python3.10/site-packages/Cython/Build/Dependencies.py", line 590, in find_pxd
[    5s]     pxd = self.context.find_pxd_file(relative, source_file_path=filename)
[    5s]   File "/usr/lib64/python3.10/site-packages/Cython/Compiler/Main.py", line 245, in find_pxd_file
[    5s]     pxd = self.search_include_directories(
[    5s]   File "/usr/lib64/python3.10/site-packages/Cython/Compiler/Main.py", line 273, in search_include_directories
[    5s]     return search_include_directories(
[    5s]   File "Cython/Utils.py", line 90, in Cython.Utils.cached_function.wrapper
[    5s]   File "/usr/lib64/python3.10/site-packages/Cython/Compiler/Main.py", line 669, in search_include_directories
[    5s]     warning(pos, "Dotted filenames ('%s') are deprecated."
[    5s]   File "/usr/lib64/python3.10/site-packages/Cython/Compiler/Errors.py", line 215, in warning
[    5s]     listing_file = threadlocal.cython_errors_listing_file
[    5s] AttributeError: '_thread._local' object has no attribute 'cython_errors_listing_file'

A full copy of the setup.py can be found at http://git.enlightenment.org/enlightenment/python-efl/src/branch/master/setup.py and a build log for openSUSE Tumbleweed can be found at https://build.opensuse.org/package/live_build_log/home:simotek:branches:X11:Enlightenment:Factory/python-efl/openSUSE_Tumbleweed/x86_64

### Code to reproduce the behaviour:

As above

### Expected behaviour

The code to comple correctly as it did with python3.8

### OS

Linux

### Python version

3.9,3.10,3.11

### Cython version

3.0.0a11

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by matusvalo**
Errors module is initialized during running pipeline:

https://github.com/cython/cython/blob/b462eec9899988a8d9c6637669ad8fcf1e3988ab/Cython/Compiler/Main.py#L521

There is a warning informing about deprecated name of .pxd file during the search of .pxd file:

https://github.com/cython/cython/blob/b462eec9899988a8d9c6637669ad8fcf1e3988ab/Cython/Compiler/Main.py#L694

However, the search can be run before running pipeline:
```
Traceback (most recent call last):
  File "/Users/matus/dev/cython/test/setup.py", line 6, in <module>
    ext_modules=cythonize("module.pyx", include_path=['bar']),
  File "/Users/matus/dev/cython/Cython/Build/Dependencies.py", line 990, in cythonize
    module_list, module_metadata = create_extension_list(
  File "/Users/matus/dev/cython/Cython/Build/Dependencies.py", line 839, in create_extension_list
    kwds = deps.distutils_info(file, aliases, base).values
  File "/Users/matus/dev/cython/Cython/Build/Dependencies.py", line 691, in distutils_info
    return (self.transitive_merge(filename, self.distutils_info0, DistutilsInfo.merge)
  File "/Users/matus/dev/cython/Cython/Build/Dependencies.py", line 700, in transitive_merge
    return self.transitive_merge_helper(
  File "/Users/matus/dev/cython/Cython/Build/Dependencies.py", line 712, in transitive_merge_helper
    for next in outgoing(node):
  File "/Users/matus/dev/cython/Cython/Utils.py", line 129, in wrapper
    res = cache[args] = f(self, *args)
  File "/Users/matus/dev/cython/Cython/Build/Dependencies.py", line 613, in cimported_files
    pxd_file = self.find_pxd(module, filename)
  File "/Users/matus/dev/cython/Cython/Utils.py", line 129, in wrapper
    res = cache[args] = f(self, *args)
  File "/Users/matus/dev/cython/Cython/Build/Dependencies.py", line 594, in find_pxd
    pxd = self.context.find_pxd_file(relative, source_file_path=filename)
  File "/Users/matus/dev/cython/Cython/Compiler/Main.py", line 270, in find_pxd_file
```

I have investigated the possibility to move `Errors` module initialisation to earlier stage but I found it rather disruptive change. Hence, I am proposing
to move the check of .pxd file name to the code called elusively within the pipeline run.

Fixes #5396

### CODE REVIEW

File: tests/errors/dotted.module.pxd
Comment (scoder) [inline tests/errors/dotted.module.pxd:1]: I'm actually not sure that this file is copied into the test directory. I think it can be deleted.

Comment (matusvalo) [reply tests/errors/dotted.module.pxd:1]: When file is deleted the test is failing.

Comment (scoder) [reply tests/errors/dotted.module.pxd:1]: Thanks for checking.

File: tests/run/dotted_filenames.srctree
Comment (scoder) [inline tests/run/dotted_filenames.srctree:1]: We already have a couple of `dotted*` tests in `build/` and `compile/`. I don't think they really test fundamentally different aspects of this feature. But it's deprecated anyway and we can possibly remove it as soon as Cython 3.1, so I don't mind adding to it.

**Review by scoder — APPROVED** [review]

**Comment by scoder** [comment]
Thanks

**Pull request merged**